code-challenges.test.js

```diff
@@ -27,8 +37,51 @@ const colors2 = [
 ]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+//------------------------------------ My Test Code ------------------------------------
+
+// pseudo code:
+// for this one I need to use the .toContain matcher to allow the return to be in any order
+// Using jest test template to test a function "chaosArray"
+// it "takes an array as an argument, removes the first item returns the rest of the array shuffled"
+// expect(chaosArray(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+// expect(chaosArray(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+// input: since there isn't a function to test, there is no input and I expect a fail (red)
+// output: should output a ReferenceError: chaosArray is not defined
+
+describe ("chaosArray", () => {
+  it("takes an array as an argument, removes the first item returns the rest of the array shuffled", () => {
+    expect(chaosArray(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+    expect(chaosArray(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+    })
+  })
+
+//output:
+// FAIL  ./code-challenges.test.js
+// ReferenceError: chaosArray is not defined
+
 // b) Create the function that makes the test pass.
 
+// pseudo code:
+// what's needed? A function "chaosArray" that takes an array as an argument, removes the first item returns the rest of the array shuffled
+// use .shift() method to remove the first item in the array
+// use array.sort((a, b) => 0.5 - Math.random()) to shuffle the array ref: https://dev.to/codebubb/how-to-shuffle-an-array-in-javascript-2ikj
+// input: an array
+// output: array shuffled, minus the first element
+// expected result:  PASS
+
+const chaosArray = (array) => {
+  array.shift()
+  return array.sort(() => Math.random() - 0.5)
+}
+
+// output:  
+// PASS  ./code-challenges.test.js
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+
+// note: I couldn't get my matcher to work by using toContain as I initially thought. I did some reaserach and found expect.arrayContaining(array) which worked ref: https://jestjs.io/docs/expect#expectarraycontainingarray
+
```

Comment:
Very clean, nice work! Also in case you didn't see, the ref you listed for arrayContaining was also given in the prompt above!

Reply:
It was?? I totally missed it LOL. Took me a while to figure out why toContain wasn't working hahaha.